src/backend/compiler/compiler.cc

```diff
@@ -990,38 +991,69 @@ extern "C" void seashell_preprocessor_free(struct seashell_preprocessor *preproc
 }
 
 /**
- * seashell_preprocessor_set_main_file(struct seashell_preprocessor* preprocessor, const char* file)
- * Sets the main file to begin running the preprocessor on
+ * seashell_preprocessor_set_question_dir(struct seashell_preprocessor* preprocessor, const char *dir)
```

Comment:
I think we need to have both for a case like the following: runner file is common/main.c, It includes common/module.h. The implementation is in the question dir, ie. q1/module.c.

One thing I could do: pass in a list of directory paths which are then added in -I flags and also searched in by the preprocessor for dependency resolution.